examples/README.md

```diff
@@ -0,0 +1,229 @@
+# Supabase `storage-php`
+
+PHP Client library to interact with Supabase Storage.
+
+## Quick Start Sample Guide
+
+### Installing the module
+
+```bash
+composer require supabase/storage-php
+```
+
+### Connecting to the storage backend
+
+```php
+include '../vendor/autoload.php';
+use Supabase\Storage\StorageClient;
+
+$authHeader = ['Authorization' => 'Bearer '.'<your_api_key>'];
+$bucket_id = 'test-bucket';
+$client = new  StorageClient(
+	'https://'.'<your_supabase_id>'.'.supabase.co/storage/v1',
+	$authHeader
+);
+```
+
+### Handling resources
+
+#### Handling Storage Buckets
+
+- Sample to create a new Storage bucket:
+
+  ```php
+    function createBucket(): void
+    {
+      global $client;
+      $result = $client->createBucket('test-bucket-3');
+      print_r($result);
+    }
+  ```
+
+- Retrieve the details of an existing Storage bucket:
+
+  ```php
+    function getBucket(): void
+    {
+      global $client;
+      $result = $client->getBucket('test-bucket');
+      print_r($result);
+    }
+  ```
+
+- Update a new Storage bucket:
+
+  ```php
+    function updateBucket(): void
+    {
+      global $client;
+      $result = $client->updateBucket('test-bucket-3', ['public' => true]);
+      print_r($result);
+    }
+  ```
+
+- Remove all objects inside a single bucket:
+
+  ```php
+    function emptyBucket(): void
+    {
+      global $client;
+      $result = $client->emptyBucket('test-bucket-2');
+      print_r($result);
+    }
+  ```
+
+- Delete an existing bucket (a bucket can't be deleted with existing objects inside it):
+
+  ```php
+    function deleteBucket(): void
+    {
+      global $client;
+      $result = $client->deleteBucket('test-bucket-3');
+      print_r($result);
+    }
+  ```
+
+- Retrieve the details of all Storage buckets within an existing project:
+
+  ```php
+    function listBuckets(): void
+    {
+      global $client;
+      $result = $client->listBuckets();
+      print_r($result);
+      foreach ($result as $bucket) {
+        print_r($bucket->name);
+      }
+    }
+  ```
+
+#### Handling Files
+
+### Connecting to the storage backend
+
+```php
+  include '../vendor/autoload.php';
+
+  use Supabase\Storage\StorageFile;
+
+  $authHeader = ['Authorization' => 'Bearer '.'<your_api_key>'];
```

Comment:
We may want to use string interpolation here:
```
$api_key = '12345';
$authHeader = ['Authorization' => "Bearer {$api_key}"];
```

This should apply to all places we are doing this in the repo.